ios/RNSScreenStack.mm

```diff
@@ -513,13 +513,6 @@ - (void)setModalViewControllers:(NSArray<UIViewController *> *)controllers
       [changeRootController dismissViewControllerAnimated:shouldAnimate completion:finish];
       return;
     }
-
-    UIViewController *lastModalVc = [self lastFromPresentedViewControllerChainStartingFrom:firstModalToBeDismissed];
-
-    if (lastModalVc != firstModalToBeDismissed) {
-      [lastModalVc dismissViewControllerAnimated:shouldAnimate completion:finish];
-      return;
-    }
```

Comment:
Have you checked if this doesn't break anything else? According to blame, this check has been here added in purpose. See: https://github.com/software-mansion/react-native-screens/commit/471127e26aa23eed4d14388a9646bf83de5e8699
Could you check if `Test1829.tsx` works properly with changes from this PR?

Reply:
Yes, we had a call with @kkafar trying to figure out the proper fix prior to making this PR and we checked a few scenarios including `Test1829`.
Unless there is some hidden scenario which breaks by this, there shouldn't be any issues with the removal of those lines.